dev/tasks/python-wheels/github.osx.yml

```diff
@@ -136,11 +141,17 @@ jobs:
       - name: Test Wheel
         env:
           PYTEST_ADDOPTS: "-k 'not test_cancellation'"
+          PYTHON_GIL: 0
         run: |
           $PYTHON -m venv test-env
           source test-env/bin/activate
           pip install --upgrade pip wheel
-          arch -{{ arch }} pip install -r arrow/python/requirements-wheel-test.txt
+          arch -{{ arch }} pip install --pre \
```

Comment:
Shouldn't this be only for free-threaded wheels? Regular wheels should work fine with PyPI-originating packages, and we want to ensure that.

Reply:
Right! I changed the step to include a conditional.